bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -92,6 +92,7 @@ static PendingAddOp create(LedgerHandle lh, ByteBuf payload, EnumSet<WriteFlag>
         op.entryId = LedgerHandle.INVALID_ENTRY_ID;
         op.currentLedgerLength = -1;
         op.payload = payload;
+        op.payload.retain();
```

Comment:
@dlg99 the change looks clear to me. but I would suggest lets add more comments around this `payload` field (line #60). basically payload should not be used anymore after it is computed to `toSend`, because the digest manager will either pass the reference count to `toSend` (for v2) and be released (by v3).